### PR TITLE
Report Filters on Citizen View

### PR DESCRIPTION
**Pull request by Type-Here**
This pull request includes significant updates to the `gestione_segnalazione_cittadino` module, focusing on improving the filtering capabilities and refactoring the code for better readability and maintainability. The most important changes include the addition of a new method to filter reports based on various criteria, reorganization of imports, and several minor refactorings to improve code clarity.

### Enhancements to Filtering Capabilities:
* Added a new method `filterReportsBy` in `CitizenReportManagementController` to filter reports based on status, priority, category, city, and date range.
* Introduced a corresponding method `filterReportsBy` in `CitizenReportManagementDAO` to handle the actual filtering logic using Firestore queries.
* `filter_model` is the main gui page for filters to be applied. A new modal was created and filter criteria are there applied, checked and restored on reopen if any.
* Filters applicable by: category, priority, status, keyword, city, date range.

### Code Refactoring:
* Reorganized imports in `gestione_segnalazione_cittadino_controller.dart` for better readability and to include necessary packages.
* Refactored the constructor and method signatures in `CitizenReportManagementDAO` for improved code clarity. [[1]](diffhunk://#diff-fc53383bd1ae3a50d08300129df8ed2ccabda7adf85bafa4ebc5f358892aecdcL24-R27) [[2]](diffhunk://#diff-fc53383bd1ae3a50d08300129df8ed2ccabda7adf85bafa4ebc5f358892aecdcL40-R44) [[3]](diffhunk://#diff-fc53383bd1ae3a50d08300129df8ed2ccabda7adf85bafa4ebc5f358892aecdcL89-R92)

This pull request includes updates to the following files:
src/lib/gestione_segnalazione_cittadino/inserimento_segnalazione_gui.dart
src/lib/gestione_segnalazione_cittadino/gestione_segnalazione_cittadino_controller.dart
src/lib/gestione_segnalazione_cittadino/visualizzazione_segnalazioni_gui.dart
src/lib/widgets/filter_modal.dart
src/lib/gestione_segnalazione_cittadino/gestione_segnalazione_cittadino_dao.dart

### Minor Improvements:
* On main added a check for browser platform for fixing issue on load up for testing purposes.
* Replaced the usage of the deprecated `category.name()` method with `category.name` in `inserimento_segnalazione_gui.dart`.
* Added new imports and state variables in `visualizzazione_segnalazioni_gui.dart` to support the enhanced filtering functionality. [[1]](diffhunk://#diff-520ffbfdd0d17c808c20837ddc42922f7077eeb46003a92f3449119f53be86edR1-R13) [[2]](diffhunk://#diff-520ffbfdd0d17c808c20837ddc42922f7077eeb46003a92f3449119f53be86edR38-R46) [[3]](diffhunk://#diff-520ffbfdd0d17c808c20837ddc42922f7077eeb46003a92f3449119f53be86edL49-R75)

These changes collectively enhance the functionality and maintainability of the citizen report management module.

### CODE REVIEW

File: src/lib/gestione_segnalazione_cittadino/gestione_segnalazione_cittadino_dao.dart [hunk 1]
```diff
@@ -99,24 +102,111 @@ Future<bool> addReport(Report reportData) async {
...
+    for (var key in criteria.keys) {
+      if (criteria[key] != null && criteria[key]!.isNotEmpty) {
+        query = query.where(key, whereIn: criteria[key]);
+      }
+    }
 
+    List<Map<String, dynamic>>? results;
+    try {
+      final querySnapshot = await query.limit(100).get(); // Check limit
+      print('QuerySnapshot: ${querySnapshot.docs.length}');
```
Comment (benedettoscala) [inline src/lib/gestione_segnalazione_cittadino/gestione_segnalazione_cittadino_dao.dart]: delete this print.
run flutter analyze before the pr.

**Review by domyz21 — APPROVED** [review]

**Review by Luigi11162 — APPROVED** [review]
LGTM! Approved

**Review by benedettoscala — APPROVED** [review]

**Review by DarkSassio — APPROVED** [review]

**Review by Peppe289 — DISMISSED** [review]
nice

File: src/lib/gestione_segnalazione_cittadino/gestione_segnalazione_cittadino_dao.dart (hunk 1 above)
Comment (Type-Here) [reply src/lib/gestione_segnalazione_cittadino/gestione_segnalazione_cittadino_dao.dart]: updated before review

**Pull request merged**